### ISSUE TITLE
[🐛] Scanning a docker image by sha256 fails (without a docker daemon)

### ISSUE THREAD

**Issue by BooleanCat** [snyk/snyk#1892]
- `node -v`:
- `npm -v`:
- `snyk -v`: `1.582.0 (standalone)`
- OS: Linux
- Command run: `snyk container test mongo@sha256:64be89e169fc33f3fa140c2c548186be4acd972bd7174344505fc5630dcf707c`

### Expected behaviour

The docker image is pulled and scanned.

### Actual behaviour

I see the following output:

```
snyk container test mongo@sha256:64be89e169fc33f3fa140c2c548186be4acd972bd7174344505fc5630dcf707c
Cannot read property 'digest' of undefined
```

### Steps to reproduce

Run `snyk container test mongo@sha256:64be89e169fc33f3fa140c2c548186be4acd972bd7174344505fc5630dcf707c` with no Docker daemon present.

### Debug log

```
# snyk -d container test mongo@sha256:64be89e169fc33f3fa140c2c548186be4acd972bd7174344505fc5630dcf707c
  snyk test <ref *1> {
  _: [
    'mongo@sha256:64be89e169fc33f3fa140c2c548186be4acd972bd7174344505fc5630dcf707c',
    [Circular *1]
  ],
  debug: true,
  docker: true,
  experimental: true
} +0ms
  snyk mongo@sha256:64be89e169fc33f3fa140c2c548186be4acd972bd7174344505fc5630dcf707c does not exist locally, proceeding to pull image. +0ms
  snyk Attempting to pull: registry: registry-1.docker.io, image: library/mongo, tag: sha256:64be89e169fc33f3fa140c2c548186be4acd972bd7174344505fc5630dcf707c +5ms
  snyk:run-test Error running test {
  error: TypeError: Cannot read property 'digest' of undefined
      at DockerPull.pull (/snapshot/snyk/node_modules/@snyk/snyk-docker-pull/dist/docker-pull.js:51:105)
      at processTicksAndRejections (internal/process/task_queues.js:93:5)
      at Docker.pull (/snapshot/snyk/node_modules/snyk-docker-plugin/dist/docker.js:28:16)
      at pullFromContainerRegistry (/snapshot/snyk/node_modules/snyk-docker-plugin/dist/analyzer/image-inspector.js:61:5)
      at pullImage (/snapshot/snyk/node_modules/snyk-docker-plugin/dist/analyzer/image-inspector.js:70:5)
      at Object.getImageArchive (/snapshot/snyk/node_modules/snyk-docker-plugin/dist/analyzer/image-inspector.js:109:9)
      at imageIdentifierAnalysis (/snapshot/snyk/node_modules/snyk-docker-plugin/dist/scan.js:70:27)
      at Object.assembleEcosystemPayloads (/snapshot/snyk/dist/lib/snyk-test/assemble-payloads.js:34:32)
      at runTest (/snapshot/snyk/dist/lib/snyk-test/run-test.js:188:26)
      at test (/snapshot/snyk/dist/cli/commands/test/index.js:71:23)
      at runCommand (/snapshot/snyk/dist/cli/index.js:44:27)
      at main (/snapshot/snyk/dist/cli/index.js:253:15)
} +0ms
  snyk-test Failed to test 1 projects, errors: +0ms
  snyk-test error: FailedToRunTestError: Cannot read property 'digest' of undefined
    at runTest (/snapshot/snyk/dist/lib/snyk-test/run-test.js:213:15)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at test (/snapshot/snyk/dist/cli/commands/test/index.js:71:23)
    at runCommand (/snapshot/snyk/dist/cli/index.js:44:27)
    at main (/snapshot/snyk/dist/cli/index.js:253:15) +0ms
Error:
Testing mongo@sha256:64be89e169fc33f3fa140c2c548186be4acd972bd7174344505fc5630dcf707c...
Cannot read property 'digest' of undefined
    at test (/snapshot/snyk/dist/cli/commands/test/index.js:178:23)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at runCommand (/snapshot/snyk/dist/cli/index.js:44:27)
    at main (/snapshot/snyk/dist/cli/index.js:253:15)
  snyk analytics {
  "args": [
    "mongo@sha256:64be89e169fc33f3fa140c2c548186be4acd972bd7174344505fc5630dcf707c",
    {
      "debug": true,
      "docker": true,
      "experimental": true
    }
  ],
  "command": "bad-command",
  "metadata": {
    "local": true,
    "error-message": "\nTesting mongo@sha256:64be89e169fc33f3fa140c2c548186be4acd972bd7174344505fc5630dcf707c...\n\nCannot read property 'digest' of undefined",
    "error": "Error: \nTesting mongo@sha256:64be89e169fc33f3fa140c2c548186be4acd972bd7174344505fc5630dcf707c...\n\nCannot read property 'digest' of undefined\n    at test (/snapshot/snyk/dist/cli/commands/test/index.js:178:23)\n    at processTicksAndRejections (internal/process/task_queues.js:93:5)\n    at runCommand (/snapshot/snyk/dist/cli/index.js:44:27)\n    at main (/snapshot/snyk/dist/cli/index.js:253:15)",
    "error-code": 500,
    "command": "test"
  },
  "version": "1.582.0",
  "os": "Linux 5.10",
  "nodeVersion": "v14.16.1",
  "standalone": true,
  "integrationName": "",
  "integrationVersion": "",
  "integrationEnvironment": "",
  "integrationEnvironmentVersion": "",
  "id": "4f43de94d141e8f20cd9ae123a9d93fd261a1362",
  "ci": false,
  "environment": {},
  "durationMs": 1481,
  "metrics": {
    "network_time": {
      "type": "timer",
      "values": [],
      "total": 0
    },
    "cpu_time": {
      "type": "synthetic",
      "values": [
        1481
      ],
      "total": 1481
    }
  }
} +0ms
  snyk Exit code: 2 +0ms
  snyk sending request to: https://snyk.io/api/v1/analytics/cli +0ms
  snyk request body size: 1208 +0ms
  snyk gzipped request body size: 576 +0ms
  snyk not using proxy +1ms
```

**Comment by snaftaly** [comment]
This issue has been fixed in version [v1.611.0](https://github.com/snyk/snyk/releases/tag/v1.611.0).
Please let us know if it can be closed.

**Comment by BooleanCat** [comment]
Thanks @snaftaly - this works for me now.

I tested by quitting Docker desktop and running `snyk container test mongo@sha256:64be89e169fc33f3fa140c2c548186be4acd972bd7174344505fc5630dcf707c`, which completed successfully.
